submissions/Ant-C-tech/dom-api-task/js/app.js

```diff
@@ -0,0 +1,881 @@
+            },
+        ],
+    },
+}
+
+//Application run
+getElements()
+recalcMap()
+addListeners()
+showContent(content.main)
+
+
+//Functions
+function getElements() {
+    mapContainer = document.querySelector('.mapContainer')
+    svg = document.querySelector('svg')
+    map = document.querySelector('.map')
+    mapAreas = document.querySelectorAll('.land')
+    hintSection = document.querySelector('#hintSection')
+    contentSection = document.querySelector('#content')
+    country = document.querySelector('#main')
+}
+
+function addListeners() {
+
+    window.onresize = recalcMap
+
+    //Add-Remove hint
+    for (const iterator of mapAreas) {
+        //Add hint on mouseenter, hover
+        iterator.addEventListener('mouseenter', function (event) {
+            if (event.target.style.fill !== 'rgb(0, 176, 255)') {
+                event.target.style.fill = '#ffeb3b'
+            }
+            hintSection.innerHTML = `${randomQuestions[getRandomIntInclusive(0, 9)]}`
+        })
+        //Remove hint on mouseleave, hover
+        iterator.addEventListener('mouseleave', function (event) {
+            if (event.target.style.fill !== 'rgb(0, 176, 255)') {
+                event.target.style.fill = '#000'
+            }
+            hintSection.innerHTML = '. . .'
+        })
+        //Change content on click
+        iterator.addEventListener('click', function (event) {
+
+            //Prohibit double click
+            if (event.target.classList.contains('show')) {
+                return false
+            } else {
+                changeContent(event)
+                event.target.classList.add('show')
+                for (const iterator of mapAreas) {
+                    if (iterator != event.target) {
+                        iterator.classList.remove('show')
+                    }
+                }
+            }
+
+        })
+    }
+
+    //Show main content on click
+    country.addEventListener('click', changeContent)
+}
+
+function recalcMap() {
+    if (document.documentElement.clientWidth >= 1040) {
+        const mapContainerWidth = mapContainer.offsetWidth
+        const mapHeight = map.getBoundingClientRect()
+        map.style.transform = `scale(${(1.35 * mapContainerWidth) / 868})`
+        svg.style.height = mapHeight.height + 'px'
+    } else {
+        map.style.transform = `scale(${(1.35 * (document.documentElement.clientWidth - 50)) / 868})`
+        const mapHeight = map.getBoundingClientRect()
+        svg.style.height = mapHeight.height + 'px'
+    }
+}
+
+function showContent(param) {
+    const content = createContent(param)
+    contentSection.appendChild(content)
+    contentSection.classList.remove('content-hide')
+}
+
+function hideContent() {
+    contentSection.classList.add('content-hide')
+}
+
+function changeContent(event) {
+    hideContent()
+    for (let item of mapAreas) {
+        item.style.fill = '#000'
+    }
+    event.target.style.fill = '#00b0ff'
+    const target = getTargetId(event)
```

Comment:
Change naming. This one is not target. It is ID.

You can get id with destructuring.

`const { id } = event.target`  or `const { id } = target` (if you can access `target` directly  )